Library/Homebrew/tap.rb

```diff
@@ -17,6 +17,9 @@ def self.clear_cache
   end
 
   def self.fetch(user, repo)
+    # we special case homebrew so users don't have to shift in a terminal
+    user = "Homebrew" if user == "homebrew"
+    repo = repo[9..-1] if repo.start_with? "homebrew-"
```

Comment:
Could the following be a worthwhile compromise between performance and legibility?

```ruby
class String
  def strip_prefix(prefix)
    start_with?(prefix) ? self[prefix.length..-1] : self
  end
end
```

It's also more reusable. On my machine, this gives:

```
Calculating -------------------------------------
           use regex    18.635k i/100ms
          use substr    37.992k i/100ms
           use strip    33.656k i/100ms
-------------------------------------------------
           use regex    272.187k (± 3.8%) i/s -      1.360M
          use substr    705.938k (± 4.5%) i/s -      3.533M
           use strip    582.990k (± 5.1%) i/s -      2.928M

Comparison:
          use substr:   705938.1 i/s
           use strip:   582990.3 i/s - 1.21x slower
           use regex:   272186.9 i/s - 2.59x slower
```